modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/EmailAddressResourceImpl.java

```diff
@@ -120,6 +163,64 @@ public Page<EmailAddress> getUserAccountEmailAddressesPage(
 				EmailAddressUtil::toEmailAddress));
 	}
 
+	@Override
+	public EmailAddress patchEmailAddress(
+			Long emailAddressId, EmailAddress emailAddress)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress =
+			_emailAddressService.getEmailAddress(emailAddressId);
+
+		return EmailAddressUtil.toEmailAddress(
+			_updateEmailAddress(serviceEmailAddress, emailAddress));
+	}
+
+	@Override
+	public EmailAddress patchEmailAddressByExternalReferenceCode(
+			String externalReferenceCode, EmailAddress emailAddress)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress =
+			_emailAddressService.fetchEmailAddressByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (serviceEmailAddress == null) {
+			throw new NoSuchEmailAddressException();
+		}
+
+		return EmailAddressUtil.toEmailAddress(
+			_updateEmailAddress(serviceEmailAddress, emailAddress));
+	}
+
+	private long _getListTypeId(String listTypeId) {
+		List<ListType> listTypes = _listTypeService.getListTypes(
+			contextCompany.getCompanyId(), listTypeId);
+
+		ListType listType = listTypes.get(0);
+
+		return listType.getListTypeId();
+	}
+
+	private com.liferay.portal.kernel.model.EmailAddress _updateEmailAddress(
+			com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress,
+			EmailAddress emailAddress)
+		throws Exception {
+
+		return _emailAddressService.updateEmailAddress(
+			GetterUtil.getString(
+				emailAddress.getExternalReferenceCode(),
+				serviceEmailAddress.getExternalReferenceCode()),
+			serviceEmailAddress.getEmailAddressId(),
+			GetterUtil.getString(
+				emailAddress.getEmailAddress(),
+				serviceEmailAddress.getAddress()),
+			GetterUtil.getLong(
+				_getListTypeId(emailAddress.getType()),
+				serviceEmailAddress.getListTypeId()),
```

Comment:
not sure about this because the emailAddress.getType I think doesn't contain the "number" (so it's not the listTypeId) but contains the "label"

Reply:
as you can see in the DTO
```
setType(
  () -> {
    ListType listType =
	serviceBuilderEmailAddress.getListType();
    return listType.getName();
});
```
we return the "label" of the type - so in input you will have the name of the type and from the name you should retrieve the id and use it in the getLong method